app/models/user.rb

```diff
@@ -2,6 +2,7 @@
 
 class User < ApplicationRecord
   devise :rememberable, :omniauthable, omniauth_providers: [:github]
+  enum role: %i[user whitelisted_user]
```

Comment:
I would just call it `admin`, not a `whitelisted_user`.
It's not like we have a whitelist _of users_ anywhere; and how do people end up in this admin list — it's an implementation detail, and might change in the future

Reply:
Another nitpick: I would recommend you to assign explicit integer values to the enum values (`enum role: {user: 0, admin: 100}`). Otherwise any accidental reshuffle of the values in the enum array will turn all regular users into admins and vice versa.

Reply:
Alternatively, you can use [Postgres database-level string enums](https://dev.to/amplifr/postgres-enums-with-rails-4ld0); but it's a little bit tricky to set up, so I would recommend you not to waste your time on that.